x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -110,6 +112,10 @@
             }
         }
     }, Setting.Property.NodeScope);
+    public static final Setting<TimeValue> DELETE_TIMEOUT = Setting.timeSetting("xpack.security.authc.api_key.delete.timeout",
+            TimeValue.MINUS_ONE, Property.NodeScope);
+    public static final Setting<TimeValue> DELETE_INTERVAL = Setting.timeSetting("xpack.security.authc.api_key.delete.interval",
```

Comment:
Since we will probably not have that many API keys expiring often, we could use a larger default value. I was thinking something in terms of few hours

Reply:
I've also been mulling over the defaults here. I think we may want to run the deletion once per day or something like that. Then maybe also have a retention longer than a day past expiration/invalidation. Maybe 1 week before having these be deleted.